api/v1/storagecluster_types.go

```diff
@@ -237,6 +237,10 @@ type ManageCephFilesystems struct {
 	// +kubebuilder:validation:MaxLength=253
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
 	StorageClassName string `json:"storageClassName,omitempty"`
+	// DataPoolSpec specifies the pool spec for the default cephfs data pool
+	DataPoolSpec rookCephv1.PoolSpec `json:"dataPoolSpec,omitempty"`
+	// AdditionalDataPoolsSpecs specifies the pool specs for additional cephfs data pools
+	AdditionalDataPoolsSpecs []rookCephv1.PoolSpec `json:"additionalDataPoolsSpecs,omitempty"`
```

Comment:
To align with the existing field (`DataPoolSpec`) I suggest we name the new field  `AdditionalDataPools` and not `AdditionalDataPoolsSpecs`
```suggestion
	DataPoolSpec rookCephv1.PoolSpec `json:"dataPoolSpec,omitempty"`
	// AdditionalDataPoolsSpecs specifies the pool specs for additional cephfs data pools
	AdditionalDataPools []rookCephv1.PoolSpec `json:"additionalDataPools,omitempty"`
```

Reply:
If we name it just AdditionalDataPools, we should change the type to []rookCephv1.NamedPoolSpec, which will have 2 fields Name & PoolSpec, https://github.com/rook/rook/blob/c409a6ae468bc9db6ad2eee1dd15e4dac9a7c03a/pkg/apis/ceph.rook.io/v1/types.go#L792-L797

Reply:
@malayparida2000 
Regardless of the name of the field, I believe we must use `rookCephv1.NamedPoolSpec` so we can refer to the data pool name when creating the storage class.  It is a good catch, as I missed that in my original review